### DISCUSSION TITLE
How can I update my github repository from my local computer?

### DISCUSSION THREAD

**Discussion by Danogbans** [EddieHubCommunity/support#7916] [General] [answered]
Hi everyone,
Hope you are all doing great.
Please, do anyone have a better explanation on how I can update/actualize an existing repository from my local computer? I stll have the project in my local machine. I have got issues with the 'git pull' command. Could anyone explain better on how and when to use 'git pull' command ? And again, it says the branches don't match. These are the processes I have applied so far:
- cd into the local repo in my machine
- git checkout (it said 'main')
- committed new changes 'git add .'
- git commit -m
- git push origin

And I was missing 'git pull'. But it said 'branches didn't  match' 
So, do anyone have a better idea on where the 'git pull' command comes in here or any other way of doing this task?
I'm using windows.
Thanks in advance!

**Comment by Panquesito7** [comment]
I believe you should update your local clone _before_ making any changes, otherwise, there will be merge conflicts.
Make sure that you cloned your repository properly, otherwise, the `git pull` command won't work.

I hope that helps. If you need any help, let us know. 🙂

**Comment by Danogbans** [comment]
Hi Panquesito,
The repository was never cloned.
This is what I am trying to do: I am trying to update a repository from my local machine where it was originally created.
I have made the changes in the original project in my local machine and now, I am trying to push these changes to the github repository in order to update the already existing project. Is there any other way I could do this or can I make the changes and commit it directly on github without cloning the repository?
- for instance, opening the repository on github, making the changes on the necessary files by editing them and committing them by pressing the commit button.

Is this possible?
Thanks

**Comment by Pradumnasaraf** [comment, marked as answer]
Hey, @Danogbans,

>I have made the changes in the original project in my local machine and now, I am trying to push these changes to the github repository in order to update the already existing project. Is there any other way I could do this or can I make the changes and commit it directly on github without cloning the repository?

Yes, you can do it. Set the remote URL in Git and push the code. Just make sure you have the same commit head, otherwise, it won't let you push.

**Comment by MAHANNAVIDU734** [comment]
Please Use GitHub Desktop and Do it well
